src/mopidy/internal/jsonrpc.py

```diff
@@ -126,18 +183,18 @@ def handle_json(self, request_json: str) -> str | None:
         :rtype: string or :class:`None`
         """
         try:
-            request: Request = json.loads(request_json, object_hook=self.decoder)
+            request = json.loads(request_json)
```

Comment:
Can't we also use `model_validate_json` here and get Pydantic's json parsing speedup ?

Reply:
That works in the happy case where the request is well-formed and can deserialize into models without problems.

I opted to just decode to native Python types here, so that we can keep all the old error handling that points out exactly what's non-conformant to the client. This is especially nice when you have a request with multiple commands, some valid and some not.

Reply:
What about chosing the fast, happy path first and then doing all this manual validation in the case of a validation error?

There's also Pydantic's [from_json](https://docs.pydantic.dev/latest/api/pydantic_core/#pydantic_core.from_json).

Reply:
Using `from_json` is a great idea. Then we get both the performance and the JSON deserialization and the model construction split into two steps, with error handling inbetween.

Reply:
I replaced `json.loads()` with `pydantic_core.from_json()` now, without any issues whatsoever.